### PR TITLE
Ava 279/http

### PR DESCRIPTION
**Pull request by kroos47**
- /v1/latest_block 
- Incase the block is not synced it returns Not Finalised message

### CODE REVIEW

File: src/http.rs [hunk 1]
```diff
@@ -1,10 +1,11 @@
 use std::{
 	net::SocketAddr,
 	str::FromStr,
-	sync::{mpsc::SyncSender, Arc},
+	sync::{mpsc::SyncSender, Arc, Mutex},
 };
 
 use anyhow::{Context, Result};
+// use async_std::channel::Receiver;
```
Comment (aterentic-ethernal) [inline src/http.rs:8]: Not needed I guess.

File: src/http.rs [hunk 2]
```diff
@@ -47,10 +53,15 @@ where
...
 	NotFound,
+	NotFinalized,
 	Error(anyhow::Error),
 }
 
-fn confidence(block_num: u64, db: Arc<DB>) -> ClientResponse<ConfidenceResponse> {
+fn confidence(
+	block_num: u64,
+	db: Arc<DB>,
+	counter: Arc<Mutex<u64>>,
```
Comment (aterentic-ethernal) [inline src/http.rs:63]: Is there an option to pass u64 value to confidence and latest_block methods, and read counter before fn call?

File: src/light_client.rs [hunk 3]
```diff
@@ -156,6 +157,8 @@ pub async fn run(
 					block_tx
 						.send(types::ClientMsg::from(params.header))
 						.context("Failed to send block message")?;
+					let mut lock = counter.lock().unwrap();
```
Comment (aterentic-ethernal) [inline src/light_client.rs:160]: Is unwrap OK or we can go with error here?

File: src/http.rs [hunk 4]
```diff
@@ -62,18 +73,41 @@ fn confidence(block_num: u64, db: Arc<DB>) -> ClientResponse<ConfidenceResponse>
...
+	res
+}
+
 fn appdata(
 	block_num: u64,
 	db: Arc<DB>,
 	cfg: &RuntimeConfig,
+	counter: Arc<Mutex<u64>>,
 ) -> ClientResponse<ExtrinsicsDataResponse> {
+	let lock = counter.lock().unwrap();
```
Comment (sh3ll3x3c) [inline src/http.rs:107]: Should we use `unwrap` here? Or some other mean of error handling? Don't know what are the consequences of not being able to acquire the lock here...

File: src/http.rs [hunk 5]
```diff
@@ -106,7 +140,14 @@ fn appdata(
 			extrinsics: data,
 		}),
 
-		Ok(None) => ClientResponse::NotFound,
+		Ok(None) => {
+			let lock = counter.lock().unwrap();
```
Comment (sh3ll3x3c) [inline src/http.rs:144]: Same...

File: src/http.rs [hunk 6]
```diff
@@ -137,22 +183,32 @@ pub async fn run_server(
...
 	_cell_query_tx: SyncSender<CellContentQueryPayload>,
+	counter: Arc<Mutex<u64>>,
 ) {
 	let host = cfg.http_server_host.clone();
 	let port = cfg.http_server_port;
 
 	let get_mode =
 		warp::path!("v1" / "mode").map(move || warp::reply::json(&Mode::from(cfg.app_id)));
+	let counter_clone = counter.clone();
+	let counter_clone1 = counter.clone();
```
Comment (sh3ll3x3c) [inline src/http.rs:194]: Do we need both of these clones, or can we just clone in place the `counter` var when passing it to the func?

File: src/http.rs (hunk 5 above)
Comment (kroos47) [reply src/http.rs:144]: changing the function to take u64 instead of Arc<>

File: src/http.rs (hunk 6 above)
Comment (kroos47) [reply src/http.rs:194]: as the values are moving in functions, I don't think without this would work! Still might be wrong

**Review by aterentic-ethernal — APPROVED** [review]
All cool, I would just rename clones to more descriptive names and that all.

**Review by sh3ll3x3c — APPROVED** [review]

**Pull request merged**